ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracingshim/__init__.py

```diff
@@ -0,0 +1,232 @@
+        # The only thing the `__init__()` method on the base class does is
+        # initialize `self._noop_span` and `self._noop_scope` with no-op
+        # objects. Therefore, it doesn't seem useful to call it.
+        # pylint: disable=super-init-not-called
+        self._tracer = tracer
+
+    @contextmanager
+    def activate(self, span, finish_on_close):
+        with self._tracer.unwrap().use_span(
+            span.unwrap(), end_on_exit=finish_on_close
+        ) as otel_span:
+            wrapped_span = SpanWrapper(
+                self._tracer, otel_span.get_context(), otel_span
+            )
+            yield ScopeWrapper(self, wrapped_span)
+
+    @property
+    def active(self):
+        span = self._tracer.unwrap().get_current_span()
+        if span is None:
+            return None
+
+        wrapped_span = SpanWrapper(self._tracer, span.get_context(), span)
+        return ScopeWrapper(self, wrapped_span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object (and the same for ScopeWrapper?).
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class TracerWrapper(opentracing.Tracer):
+    def __init__(self, tracer, scope_manager=None):
+        # If a scope manager isn't provided by the user, create a
+        # `ScopeManagerWrapper` instance and use it to initialize the
+        # `TracerWrapper`.
+        if scope_manager is None:
+            scope_manager = ScopeManagerWrapper(self)
+        super().__init__(scope_manager=scope_manager)
+        self._otel_tracer = tracer
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Tracer` object."""
+
+        return self._otel_tracer
+
+    @contextmanager
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ):
+        span = self.start_span(
+            operation_name=operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        with self._scope_manager.activate(span, finish_on_close) as scope:
+            yield scope
+
+    def start_span(
+        self,
+        operation_name=None,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+    ):
+        parent = child_of
+        if parent is not None:
+            if isinstance(parent, SpanWrapper):
+                parent = child_of.unwrap()
+            elif isinstance(parent, SpanContextWrapper):
+                parent = child_of.unwrap()
+            else:
+                raise RuntimeError(
+                    "Invalid parent type when calling start_span()."
+                )
```

Comment:
It's a problem for another PR, but we'll probably want to log the error and return a no-op span instead of throwing here.

Reply:
Yes @c24t,  at some point I've realized we probably don't want to break the application if we can't trace something.
Following [this](https://github.com/open-telemetry/opentelemetry-python/pull/211#discussion_r333468806) comment from @Oberon00 I thought about logging a warning. The question is - **what then**?
- Refuse to create a span and just return `None` from `start_span()`?
- Create a span with no parent so that it either becomes a child of the active span or forms a new trace?